web/app/charts/MapChart.js

```diff
@@ -51,6 +58,53 @@ bluewave.charts.MapChart = function(parent, config) {
         }
 
         mapArea = svg.append("g");
+        svg.call(zoom);
+    };
+
+  //**************************************************************************
+  //** handleZoom
+  //**************************************************************************
+    function handleZoom(){
+        var projection = me.getProjection();
+        mapArea.attr('transform', d3.event.transform);
+        if (projection){
+            var rect = javaxt.dhtml.utils.getRect(svg.node());
+            var w = rect.width;
+            var h = rect.height;
+            var t = d3.event.transform;
+            var x = (w/2)-t.x;
+            var y = (h/2)-t.y;
+            var p = projection.invert([x,y]);
+            centeringConfig.newCenter = p;
```

Comment:
Can we get rid of the centeringConfig and just update the chartConfig directly?

Reply:
Not really easily, chartConfig is only passed into the update methods. Which means that the handleZoom has no access to it..

Reply:
I will see what I can do though.

Reply:
Per our conversation in Slack I have moved all of this over to ChartConfig